unit-tests/utilities/concurrency/test-dispatcher.cpp

```diff
@@ -95,3 +96,40 @@ TEST_CASE("verify stop() not consuming high CPU usage")
     // We allow 9 seconds to support debug configuration as well
     REQUIRE(sw.get_elapsed_ms() < 9000);
 }
+
+TEST_CASE("stop() notify flush to finish")
+{
+    // On this test we check that if during a flush() another thread call stop(),
+    // than the flush CV will be triggered to exit and not wait a full timeout
+    dispatcher dispatcher( 10 );
+    dispatcher.start();
+
+    stopwatch sw;
+    std::atomic_bool dispatched_end_verifier{ false };
+    dispatcher.invoke( [&]( dispatcher::cancellable_timer c ) {
+        //std::cout << "Sleeping from inside invoke" << std::endl;
+        std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
+        //std::cout << "Sleeping from inside invoke - Done" << std::endl;
+        dispatched_end_verifier = true;
+    } );
+
+    // Make sure the above invoke function is dispatched
+    std::this_thread::sleep_for( std::chrono::milliseconds( 100 ) );
+
+    std::thread t( [&]() {
```

Comment:
Rename, please: `stop_thread` is better than `t`

Reply:
👍